ex/high_throughput_inference/mock_app.py

```diff
@@ -0,0 +1,154 @@
+        self._from_worker_ch_serialized = self._from_worker_ch.serialize()
+        self._to_worker_ch = Channel.make_process_local()
+
+        self._start = None
+        self._interm = None
+        self._timings: OrderedDict[str, list[numbers.Number]] = OrderedDict()
+        self._timing_on = timing_on
+
+    def _add_label_to_timings(self, label: str):
+        if label not in self._timings:
+            self._timings[label] = []
+
+    @staticmethod
+    def _format_number(number: numbers.Number):
+        return f"{number:0.4e}"
+
+    def start_timings(self, batch_size: int):
+        if self._timing_on:
+            self._add_label_to_timings("batch_size")
+            self._timings["batch_size"].append(batch_size)
+            self._start = time.perf_counter()
+            self._interm = time.perf_counter()
+
+    def end_timings(self):
+        if self._timing_on:
+            self._add_label_to_timings("total_time")
+            self._timings["total_time"].append(self._format_number(time.perf_counter()-self._start))
+
+    def measure_time(self, label: str):
+        if self._timing_on:
+            self._add_label_to_timings(label)
+            self._timings[label].append(self._format_number(time.perf_counter()-self._interm))
+            self._interm = time.perf_counter()
+
+    def print_timings(self, to_file: bool = False):
+        print(" ".join(self._timings.keys()))
+        value_array = numpy.array([value for  value in self._timings.values()], dtype=float)
+        value_array = numpy.transpose(value_array)
+        for i in range(value_array.shape[0]):
+            print(" ".join(self._format_number(value) for value in value_array[i]))
+        if to_file:
+            numpy.save("timings.npy", value_array)
+            numpy.savetxt("timings.txt", value_array)
+
+
+    def run_model(self, model: bytes, batch: torch.Tensor):
+        self.start_timings(batch.shape[0])
+        built_tensor = MessageHandler.build_tensor(
+            batch.numpy(), "c", "float32", list(batch.shape))
+        self.measure_time("build_tensor")
+        request = MessageHandler.build_request(
+            reply_channel=self._from_worker_ch_serialized,
+            model=model,
+            inputs=[built_tensor],
+            outputs=[],
+            output_descriptors=[],
+            custom_attributes=None,
+        )
+        self.measure_time("build_request")
+        request_bytes = MessageHandler.serialize_request(request)
+        self.measure_time("serialize_request")
+        with self._to_worker_fli.sendh(timeout=None, stream_channel=self._to_worker_ch) as to_sendh:
+            to_sendh.send_bytes(request_bytes)
+
+        self.measure_time("send")
+        with self._from_worker_ch.recvh(timeout=None) as from_recvh:
+            resp = from_recvh.recv_bytes(timeout=None)
+            self.measure_time("receive")
+            response = MessageHandler.deserialize_response(resp)
+            self.measure_time("deserialize_response")
```

Comment:
Quick question. Shouldn't we be deserializing the request here? We serialize the request but I don't see where it's deserialized.

Reply:
Unless that's actually measured during`app_receive`, and if so never mind!

Reply:
I think on the app side we only want to deserialize the response - is there something I'm missing?